### ISSUE TITLE
Demonstrate how to crop by column and row number

### ISSUE THREAD

**Issue by Robinlovelace** [Robinlovelace/geocompr#277]
Discovered how to do this preparing for geostat and I think it's worth mentioning:

``` r
library(raster)
#> Loading required package: sp
r1 = r2 = raster(ncols = 3, nrows = 3)
values(r1) = 1:9
plot(r1)
values(r2) = 9:1
r_cropped = crop(r2, extent(r2, 1, 1, 1, 2))
plot(r1)
plot(r_cropped, add = T)
```

![](https://i.imgur.com/lRDa1wp.png)

Created on 2018-08-18 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by Nowosad** [comment]
Alternatively, you can use cell numbers:

``` r
library(raster)
#> Loading required package: sp
#> Loading required package: sp
r1 = r2 = raster(ncols = 3, nrows = 3)
values(r1) = 1:9
plot(r1)
values(r2) = 9:1
r_cropped = crop(r2, extent(r2, 1, 1, 1, 2))
plot(r1)
plot(r_cropped, add = TRUE)
```

![](https://i.imgur.com/hE9BLoD.png)

``` r

# alternative approach ----------------------------------------------------
r_cropped2 = crop(r2, extentFromCells(r2, c(1, 2)))
identical(r_cropped, r_cropped2)
#> [1] TRUE
```

Created on 2018-08-21 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by Robinlovelace** [comment]
Great, they are equally complex methods. We need to include at least one of them before closing this issue. I suggest the first approach is slightly easier because `extent()` is a more commonly used function. Suggest we add a demo with that in it and then mention `extentFromCells()` in the prose.

**Comment by Nowosad** [comment]
Yep. The only question is how to fit this example to the text.

**Comment by Nowosad** [comment]
Hm.. I don't know what's the best place for it. @jannes-m ?

**Comment by jannes-m** [comment]
Mmh, not sure, isn't this a bit overly complex? I mean you are simply subsetting the first two values of a raster, and for this you use the `extent()` command (or am I missing something?) which is not really intuitive imo (in this respect I would like Jakub's approach better). Nevertheless, wouldn't it be much easier just to do:

```r
r2[1:2, drop = FALSE]
```
And done. This is already covered in the section you mention.

**Comment by Robinlovelace** [comment]
@jannes-m that answers my question is much cleaner. `drop = FALSE` is mentioned but only in the context of using another object as the first argument in `[` in `elev[rmask, drop = FALSE]`. I think it's more basic to say this is how you subset by columns and rows (I wish I had this info 2 days ago, an example when I used the book to solve a problem):

``` r
library(raster)
#> Loading required package: sp
r1 = r2 = raster(ncols = 3, nrows = 3)
values(r1) = 1:9
plot(r1)
```

![](https://i.imgur.com/G0TEuzE.png)

``` r
values(r2) = 9:1
r1[1:2, 1:2]
#> [1] 1 2 4 5
r1[1:2, 1:2, drop = FALSE]
#> class       : RasterLayer 
#> dimensions  : 2, 2, 4  (nrow, ncol, ncell)
#> resolution  : 120, 60  (x, y)
#> extent      : -180, 60, -30, 90  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0 
#> data source : in memory
#> names       : layer 
#> values      : 1, 5  (min, max)
r1[1:2, 1:2, drop = FALSE][]
#> [1] 1 2 4 5
r3 = r2[1:2, 1:2, drop = FALSE]
r3[]
#> [1] 9 8 6 5
```

Created on 2018-08-21 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by jannes-m** [comment]
This is covered in https://geocompr.robinlovelace.net/attr.html#raster-subsetting but I guess you are right and we should add to the spatial subsetting section that you can also use cell numbers or row and column numbers for spatial subsetting as well by adding `drop = FALSE`. And now I am going to Prague, it's much easier to discuss this stuff in person. Really looking forward to meeting you guys!

**Comment by jannes-m** [comment]
Closed via 55d8739. @Robinlovelace Have a look if you are happy with the changes! In any case it is a good idea to explicitly show how spatial raster subsetting also returns a spatial object. Thanks!

**Comment by jannes-m** [comment]
Thanks for making the code more concise. You are also right in pointing out that `raster()` full parameter names are `nrows` and `ncols`, I have changed this now. Additionally, there was in fact something wrong with the `mask()` code, I have fixed this now. 
I have replaced `identical()` by `all.equal()` and everything is fine. Not sure, why `identical()` returns `FALSE` when in fact the exact same raster objects are returned.
